test/e2e/framework/util.go

```diff
@@ -615,8 +615,9 @@ func CoreDump(dir string) {
 		Logf("Dumping logs locally to: %s", dir)
 		cmd = exec.Command(path.Join(TestContext.RepoRoot, "cluster", "log-dump", "log-dump.sh"), dir)
 	}
-	cmd.Env = append(os.Environ(), fmt.Sprintf("LOG_DUMP_SYSTEMD_SERVICES=%s", parseSystemdServices(TestContext.SystemdServices)))
-	cmd.Env = append(os.Environ(), fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
+	env := os.Environ()
+	env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_SERVICES=%s", parseSystemdServices(TestContext.SystemdServices)))
+	cmd.Env = append(env, fmt.Sprintf("LOG_DUMP_SYSTEMD_JOURNAL=%v", TestContext.DumpSystemdJournal))
```

Comment:
so the original code does this:
A = ENV + X
A = ENV + Y

so A ends up being ENV + Y and X is ignored.
the mistake is from https://github.com/kubernetes/kubernetes/commit/35543f89891ff4558010d4c6adc2ac08475e45f5

of course X in this case is just adding dump of systemd logs too., so there shouldn't be a problem as long as parseSystemdServices is OK.

CI is running on this PR, so lets see if everything passes.

Reply:
This bug was introduced in 2019 and no-one noticed until now? Can we perhaps remove and/or simplify this code instead of fixing it?

@jbtk: how did you come across this?

Reply:
I was trying to run autoscaler tests with kubetest2 which happens no to pass the env vars needed for dumping the logs after the test. Because there are a few layers of things that get called I was hardcoding env variables in callsites to see what is needed (and later figure out which flags and how to set). I came across this call site and tried adding another line and figured out it does not work as expected and realized that there is a problem. I have no experience with this codebase so decided to fix rather than remove, but I am happy to change it and remove if we decide that this is a better way to go.

Reply:
Thanks for taking the time to investigate and sharing the result.

Let's merge, although I am still wondering whether we really need this feature.

/approve